paddle/framework/channel.h

```diff
@@ -50,6 +51,11 @@ Channel<T>* MakeChannel(size_t buffer_size) {
 
 template <typename T>
 void CloseChannel(Channel<T>* ch) {
+  ch->Close();
+}
+
+template <typename T>
+void DeleteChannel(Channel<T>* ch) {
```

Comment:
I agree that we need CloseChannel, which mimics Go's close, and DeleteChannel, which mimics Go's garbage collection.

One step forward, I think we can have only CloseChannel, because DeleteChannel could be simplified as `delete ch` -- I didn't have to and should not have defined `template <typename T> void DeleteChannel`.

Reply:
I agree with you. 
We should make a further analysis about `CloseChannel`. When the channel was closed, the `channel_`  of `Channel` may also have some data, I think that `receive` can also work properly, but `send` cannot send data to `channel_`. 
So, `DeleteChannel` should not delete `Buffered` or `Unbuffered`.